system/ee/ExpressionEngine/Model/Content/ContentModel.php

```diff
@@ -503,6 +503,28 @@ protected function deleteFieldData()
                 ->delete($tables);
         }
     }
+
+    /**
+     * Mark a property or the entire entity as dirty.
+     *
+     * @param String $name Property name [optional]
+     */
+    public function markAsDirty($name = null)
+    {
+        if (!empty($name)) {
+            if (! $this->hasBackup($name) && $this->hasCustomField($name)) {
+                $this->setBackup($name, $this->getRawProperty($name));
+            }
+        } else {
+            foreach ($this->getCustomFieldNames() as $field) {
+                $this->setBackup($field, $this->getRawProperty($field));
+            }
+        }
+
+        parent::markAsDirty($name);
+
+        return $this;
+    }
```

Comment:
This looks like substantially the same code that was added in system/ee/ExpressionEngine/Library/Data/Entity.php in this PR. Can we reuse that? tagging @intoeetive

Reply:
It is not exactly same. This code is also marking entry's custom fields as "dirty" which parent function (also called here) can't do